en/contributing/plugins.md

```diff
@@ -68,6 +68,54 @@ The [standard plugins](https://github.com/dronecore/DroneCore/tree/{{ book.githu
 how to write plugin code, including how to send and process MAVLink messages.
 
 
+### Plugin Enable/Disable
+
+DroneCore provides virtual methods that a plugin should implement allow DroneCore to better manage resources. For example, to prevent callback being created before the `Device` is instantiated, or messages being sent when a vehicle is not connected.
+
+Plugin authors should provide an implementation class that derives from `PluginImplBase` (**core/plugin_impl_base.h**) and implements the following pure virtual methods:
+* [init()](#init)/[deinit()](#deinit): These are called when a device is created and just before it is destroyed. These should be used for setting up and cleaning everything that depends on having the `Device` instantiated. This includes calls that set up callbacks.
+* [enable()](#enable)/[disable()](#disable): These are called when a vehicle is discovered or has timed out. They should be used for managing resources needed to access a connected device/vehicle (e.g. getting a parameter or changing a setting).
+
+
+> **Note** The methods are currently optional. For a transitional period, runtime warnings will be displayed if the methods are not implemented.
+
+Additional detail is provided for methods below.
+
+
+##### init() {#init}

```

Comment:
I pulled in your docs here because the PluginImplBase  is not part of the public API, and hence not automatically exported and build into the API Reference. 
Should it be? How can we make that happen?

IF that happens I'd remove these docs here and link to the ref. That would be better.

Reply:
This will change in the future as dronecore becomes it's own library that libraries link to. At that point there will be a public API for it to be used by plugin developers.

Reply:
OK. I'll update that as a PR when those changes go in.